### DISCUSSION TITLE
Memory error when cloning a tiff image

### DISCUSSION THREAD

**Discussion by ghost** [dlemstra/Magick.NET#681] [Help]
Hello,

One specific tiff image is crashing our code, 
while it is working ok with any other images (other tiff images and other formats as well)

rogue image:  https://www.dropbox.com/s/97jfpi2m9l2a7f1/a239jy.tif?dl=0

crashing code:

`
 // if multiple images in collection take the one with alpha layer
 if (magickImageCollection.Count > 1) {
  image = (MagickImage)magickImageCollection.FirstOrDefault(x => x.HasAlpha);
}

 // in case we didn't have a frame with alpha
  if (image == null) {
   image = (MagickImage)magickImageCollection.Flatten(new MagickColor(MagickColors.Transparent));
}

`

the magickImageCollection.Flatten is returning a weird image

It looks like this specific tiff has only one frame in the collection, and thus i added this code to extract the image

`
if (magickImageCollection.Count == 1)
 {
  image = (MagickImage)magickImageCollection.FirstOrDefault();
 }
`

Can someone explain me what is happening there in the magickImageCollection.Flatten with this tiff ?

we use magickImageCollection.Flatten to extract all images,
without problem even when the collection has only one frame

Thanks,

Luc

**Comment by dlemstra** [comment]
Can you explain in more detail what kind of issue you have? I also cannot reproduce it.
